include/mrdox/Config.hpp

```diff
@@ -42,25 +42,59 @@ class MRDOX_DECL
     Config() noexcept;
 
 public:
-    struct Settings
+    struct ExtractOptions
     {
-        /** `true` if anonymous namespace members should be extracted and displayed.
+        enum class Policy
+        {
+            Always,
+            Dependency,
+            Never
+        };
+
+        /** Extraction policy for references to external declarations.
+
+            This determines how declarations which are referenced by
+            explicitly extracted declarations are extracted.
+
+            Given a function parameter of type `std::string`, `std::string`
+            would be extracted if this option is set to `Policy::Always`.
+        */
+        Policy referencedDeclarations = Policy::Dependency;
+
+        /** Extraction policy for anonymous namespace.
+
+            @li `Policy::Always`: anonymous namespaces and their
+            members will always be extracted.
+
+            @li `Policy::Dependency`: members of anonymous namespaces will only
+            be extracted via dependency.
 
-            In some cases anonymous namespace members will
-            be listed even if this configuration value is set to
-            `false`. For example, this may occur for a class derived
-            from one declared within an anonymous namespace.
+            @li `Policy::Never`: members of anonymous namespace will
+            never be extracted, regardless of how they are referenced.
         */
-        bool includeAnonymous = true;
+        Policy anonymousNamespaces = Policy::Always;
 
-        /** `true` if private members should be extracted and displayed.
+        /** Extraction policy for inaccessible members.
+
+            @li `Policy::Always`: all `private` and `protected` members
+            will be extracted.
+
+            @li `Policy::Dependency`: `private` and `protected` members will only
+            be extracted via dependency.
+
+            @li `Policy::Never`: `private` and `protected` will never be extracted.
+        */
+        Policy inaccessibleMembers = Policy::Always;
+
+        Policy inaccessibleBases = Policy::Always;
+    };
```

Comment:
I'm not sure how I feel about creating a deep hierarchy of settings structs. Can't we just leave it flat? e.g.
```
Policy extractInaccessibleMembers;
Policy inaccessibleBases;
```
And so on?

Reply:
I personally prefer nesting because it reduces "config sprawl", especially once we add more of these extraction options. However, if you feel strongly about this I can flatten it.

Reply:
These structs aren't used in the plugins, and should probably not be public

Reply:
> if you feel strongly about this I can flatten it

I would leave it flat and only make it hierarchical when there is a well demonstrated benefit.